dffml/source/file.py

```diff
@@ -65,7 +65,7 @@ class FileSource(BaseSource):
         elif self.config.filename[::-1].startswith((".zip")[::-1]):
             opener = self.zip_opener_helper()
         else:
-            opener = open(self.config.filename, "r")
+            opener = open(self.config.filename, "rb")
```

Comment:
```python
opener = open(self.config.filename, self.READ_MODE)
```

then subclasses can override the READ_MODE and WRITE_MODE, we'll also need READ_MODE_COMPRESSED and WRITE_MODE_COMPRESSED

Reply:
Create a new subclass BinaryFileSource with those properties set, then have IDX* source subclass from BinaryFileSource